src/ui/win32/DialogBase.cpp

```diff
@@ -20,47 +22,58 @@ DialogBase::~DialogBase() noexcept
     }
 }
 
-static INT_PTR CALLBACK StaticDialogProc(HWND hDlg, UINT uMsg, WPARAM wParam, LPARAM lParam)
+_Use_decl_annotations_
+_NODISCARD static INT_PTR CALLBACK StaticDialogProc(HWND hDlg, UINT uMsg, WPARAM wParam, LPARAM lParam)
 {
-    DialogBase* pDialog = reinterpret_cast<DialogBase*>(GetWindowLongPtr(hDlg, DWLP_USER));
+    const auto pDialog{ reinterpret_cast<DialogBase*>(GetWindowLongPtr(hDlg, DWLP_USER)) };
+
+    // TBD: A dialog should not be calling DefWindowProc, it should return 0 
     if (pDialog == nullptr)
-        return DefWindowProc(hDlg, uMsg, wParam, lParam);
+        return ::DefWindowProc(hDlg, uMsg, wParam, lParam);
 
-    INT_PTR result = pDialog->DialogProc(hDlg, uMsg, wParam, lParam);
+    const auto result{ pDialog->DialogProc(hDlg, uMsg, wParam, lParam) };
 
     if (uMsg == WM_DESTROY)
-        SetWindowLongPtr(hDlg, DWLP_USER, 0L);
+        ::SetWindowLongPtr(hDlg, DWLP_USER, 0L);
 
     return result;
 }
 
-HWND DialogBase::CreateDialogWindow(LPTSTR sResourceId, IDialogPresenter* pDialogPresenter)
+_Use_decl_annotations_
+HWND DialogBase::CreateDialogWindow(const LPCTSTR sResourceId, IDialogPresenter* const pDialogPresenter)
 {
-    m_hWnd = CreateDialog(g_hThisDLLInst, sResourceId, g_RAMainWnd, &StaticDialogProc);
+    m_hWnd = ::CreateDialogParam(g_hThisDLLInst, sResourceId, g_RAMainWnd, StaticDialogProc,
+                               reinterpret_cast<LPARAM>(this));
```

Comment:
There's no reason for this change. The additional value passed to WM_INITDIALOG is completely ignored.

And how does this fit in with the three rules you've identified as the target of this PR?

Reply:
Sorry, it'll go back.